src/Microsoft.AspNet.OData.Shared/Query/Validators/ODataQueryValidator.cs

```diff
@@ -62,15 +64,22 @@ public virtual void Validate(ODataQueryOptions options, ODataValidationSettings
                 options.Filter.Validate(validationSettings);
             }
 
-            if (options.Count != null || options.InternalRequest.IsCountRequest())
+            if (options is ODataQueryOptions _options)
```

Comment:
Since we have moved to the Interface shouldn't we allow custom implementations of IODataQueryOptions to work as with this current implementation custom implementations are currently locked out.

Reply:
InternalRequest does not exist on the interface because it is access is internal. on line 69, can || _options.InternalRequest.IsCountRequest() be factored out?